dspace-server-webapp/src/main/java/org/dspace/app/rest/converter/ConverterService.java

```diff
@@ -52,9 +53,12 @@
 /**
  * Converts domain objects from the DSpace service layer to rest objects, and from rest objects to resource
  * objects, applying {@link Projection}s where applicable.
- *
+ * <P>
+ * MUST be loaded @Lazy, as this service requires other services to be preloaded (especially DSpaceConverter components)
+ * and that can result in circular references if those services need this ConverterService (and many do).
  * @author Luca Giamminonni (luca.giamminonni at 4science dot it)
  */
+@Lazy
```

Comment:
@LucaGiamminonni : I suspect the problem only occurs now because the new version of Spring is more "strict" about circular dependencies.  This problem *did* exist already (and has been there since 7.0 I believe).  It's just that Spring ignored it before...while the new version of Spring throws circular dependency errors.

I can look into whether a `@PostContruct` method is easy to implement here.  If so, I'll create it.  However, if it's more complex than that, I think we'll need to leave this PR "as-is" and fix it in a follow-up PR.  I don't want to delay our upgrade to the latest Spring.

Reply:
@LucaGiamminonni : I tried this today, and it's not as easy as it appears.  `ConverterService` already has a `@PostContruct` method. While I *can* move the `DSpaceConverter` classes to that method & get them to work... there's a major issue between `ConverterService` and `Utils` (both of them `@Autowired` the other) which doesn't seem possible to workaround simply without a possible larger refactor.

I'll dig around a bit more tomorrow, but this might need to wait for a separate PR.  There's too many circular dependencies that look to be "hidden" on `main` right now...upgrading to latest Spring makes them all "visible" (in that I have to add `@Lazy` to all of them). But, it doesn't appear they are all easy to cleanup.   At the very least, I might be able to cleanup a few that are easier / obvious (to avoid a few `@Lazy` annotations), but some may still remain.

Reply:
I gave this another try, and have had no success. I think this requires a larger refactor, as there are a lot of pieces which need rework.  While I can get things to compile/deploy by moving `DSpaceConverter` classes into the `@PostContruct` method, they don't work -- I experience a lot of runtime failures via that route.  The core issue appears to be that currently `DSpaceConverter` classes depend heavily on `ConverterService`, and `ConverterService` similarly depends on every `DSpaceConverter`. This interdependency isn't fixed by using a `@PostConstruct` method (or at least I've not been able to figure out a way to do so).

The only stable "quick fix" appears to be these `@Lazy` annotations, as that essentially gets the existing (flawed) code to work in the latest version of Spring.  Beyond that will require refactors that seem out-of-scope for this PR (and has been logged in #8233 instead)